dbms/src/Storages/DeltaMerge/Segment.cpp

```diff
@@ -3173,21 +3171,16 @@ BlockInputStreamPtr Segment::getLateMaterializationStream(
         segment_snap->delta,
         rest_columns_to_read,
         this->rowkey_range);
-    SkippableBlockInputStreamPtr rest_column_stream = std::make_shared<RowKeyOrderedBlockInputStream>(
-        *rest_columns_to_read,
-        rest_column_stable_stream,
-        rest_column_delta_stream,
-        segment_snap->stable->getDMFilesRows(),
-        dm_context.tracing_id);
-
+    auto rest_column_stream
+        = std::dynamic_pointer_cast<ConcatSkippableBlockInputStream<false>>(rest_column_stable_stream);
+    rest_column_stream->appendChild(rest_column_delta_stream, segment_snap->delta->getRows());
```

Comment:
Use `RUNTIME_CHECK` to check if `rest_column_stream` is null for potential cast failure?